model/testutil/stats.go

```diff
@@ -0,0 +1,112 @@
+package testutil
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func createTestStatsId(project string, requester string, testFile string, taskName string,
+	variant string, distro string, date time.Time) bson.D {
+	return bson.D{
+		{Name: "test_file", Value: testFile},
+		{Name: "task_name", Value: taskName},
+		{Name: "variant", Value: variant},
+		{Name: "distro", Value: distro},
+		{Name: "project", Value: project},
+		{Name: "requester", Value: requester},
+		{Name: "date", Value: date},
+	}
+}
+
+func createTaskStatsId(project string, requester string, taskName string, variant string, distro string, date time.Time) bson.D {
+	return bson.D{
+		{Name: "task_name", Value: taskName},
+		{Name: "variant", Value: variant},
+		{Name: "distro", Value: distro},
+		{Name: "project", Value: project},
+		{Name: "requester", Value: requester},
+		{Name: "date", Value: date},
+	}
+}
+
+func getTestStatsDoc(s suite.Suite, collection string, project string, requester string,
+	testFile string, taskName string, variant string, distro string, date time.Time) *dbTestStats {
+	doc := dbTestStats{}
+	docId := createTestStatsId(project, requester, testFile, taskName, variant, distro, date)
+	err := db.FindOne(collection, bson.M{"_id": docId}, db.NoProjection, db.NoSort, &doc)
+	if err == mgo.ErrNotFound {
+		return nil
+	}
+	s.Require().NoError(err)
+	return &doc
+}
+
+func GetDailyTestDoc(s suite.Suite, project string, requester string, testFile string,
+	taskName string, variant string, distro string, date time.Time) *dbTestStats {
+	return getTestStatsDoc(s, "daily_test_stats", project, requester, testFile, taskName,
+		variant, distro, date)
+}
+
+func GetHourlyTestDoc(s suite.Suite, project string, requester string, testFile string,
+	taskName string, variant string, distro string, date time.Time) *dbTestStats {
+	return getTestStatsDoc(s, "hourly_test_stats", project, requester, testFile, taskName,
+		variant, distro, date)
+}
+
+func GetDailyTaskDoc(s suite.Suite, project string, requester string, taskName string, variant string, distro string, date time.Time) *dbTaskStats {
+	doc := dbTaskStats{}
+	docId := createTaskStatsId(project, requester, taskName, variant, distro, date)
+	err := db.FindOne("daily_task_stats", bson.M{"_id": docId}, db.NoProjection, db.NoSort, &doc)
+	if err == mgo.ErrNotFound {
+		return nil
+	}
+	s.Require().NoError(err)
+	return &doc
+}
+
+////////////////////////////////////////
+// Structs to represent database data //
+////////////////////////////////////////
+
+type dbTestStatsId struct {
```

Comment:
This was originally parts of Yves test code for testing the stats model. I've just moved it to a testutil package so that I could use it as well. I don't see another copy of these elsewhere. I know Yves is working on improving the stats model some right now.